### PR TITLE
Enforce maximum length in entity instance ids

### PR DESCRIPTION
**Pull request by halspang**
# Summary
## What changed?
- Enforce maximum length of entity instances

## Why is this change needed?
- The backend expects a maximum length of 100 characters for an Entity's ID. This change enforces that at the SDK level, allowing for an opt-in breaking change to eliminate bad behavior on the backend.

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [ ] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [ ] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [ ] All required tests have been added/updated (unit tests, E2E tests)
- [x] Breaking change?
    - Impact: Users will no longer be able to specify Entity Instance Ids (combination of name and key) longer than 100 characters
    - Migration guidance: Use smaller name/key combos for Entities. If entities already exist with IDs that are too long, you must use the truncated value to access them: `entityInstanceId.ToString().Take(100)`
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [x] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s):
- AI-assisted areas/files:
- What you changed after AI output:

AI verification (required if AI was used):
- [ ] I understand the code and can explain it
- [ ] I verified referenced APIs/types exist and are correct
- [ ] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [ ] I reviewed concurrency/async behavior
- [ ] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed

---

# Notes for reviewers
- N/A

### CODE REVIEW

File: src/Abstractions/Entities/EntityInstanceId.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ public EntityInstanceId(string name, string key)
         Check.NotNull(key);
         this.Name = name.ToLowerInvariant();
         this.Key = key;
+
+        if (this.ToString().Length > 100)
+        {
+            throw new ArgumentException("entity instance ids may not exceed 100 characters in length.");
+        }
```
[the remark this answers is not in the document]
Comment (halspang) [reply src/Abstractions/Entities/EntityInstanceId.cs:38]: We want to validate `ToString()` because it's the actual behavior to test. Having duplicate behavior over here just gives us places to diverge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 - Change the default large payload externalization threshold (`LargePayloadStorageOptions.ThresholdBytes`) from 900,000 bytes to 256 KiB (262,144 bytes)
+- **BREAKING**: `EntityInstanceId` now limits total length to 100 characters to match backend behavior
```
[the remark this answers is not in the document]
Comment (halspang) [reply CHANGELOG.md]: Fair.

File: src/Abstractions/Entities/EntityInstanceId.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (halspang) [reply src/Abstractions/Entities/EntityInstanceId.cs]: It is opt-in in that they control upgrading the dependency, they cannot disable this behavior.
